File: threat-detection-platform/backend/app/schemas/alert.py
"""
Alert Schemas
"""

from typing import Optional

from pydantic import BaseModel, Field


class AlertResponse(BaseModel):
    """Alert detail response."""

    id: str
    alert_number: int = 0
    incident_id: str
    camera_id: str
    severity: str
    title: str
    message: str
    status: str = "PENDING"
    is_acknowledged: bool = False
    acknowledged_by: str | None = None
    acknowledged_at: str | None = None
    acknowledgement_notes: str | None = None
    dismissed_by: str | None = None
    dismissed_at: str | None = None
    created_at: str
    updated_at: str | None = None

    model_config = {"from_attributes": True}


class NotificationLogResponse(BaseModel):
    """A single notification delivery attempt record."""

    id: str
    channel: str
    status: str
    attempt_number: int
    max_attempts: int
    error_message: str | None = None
    created_at: str
    sent_at: str | None = None
    next_retry_at: str | None = None


class AlertAcknowledgeRequest(BaseModel):
    """Acknowledge an alert."""

    notes: Optional[str] = Field(default=None, max_length=500)


class AlertDismissRequest(BaseModel):
    """Dismiss an alert without treating it as an acknowledged threat."""

    reason: Optional[str] = Field(default=None, max_length=500)


class AlertCountResponse(BaseModel):
    """Alert counts for dashboard badge."""

    total_unread: int
    total_unacknowledged: int
    by_severity: dict = {}

File: threat-detection-platform/frontend/src/types/alert.ts
import { Severity } from './common';

export interface Alert {
  id: string;
  incident_id: string;
  camera_id: string;
  severity: Severity;
  title: string;
  message: string;
  alert_type: string;
  is_read: boolean;
  is_acknowledged: boolean;
  acknowledged_by: string | null;
  acknowledged_at: string | null;
  created_at: string;
}

export interface AlertCounts {
  total_unread: number;
  total_unacknowledged: number;
  by_severity: Record<string, number>;
}
